lib/cs_guide/resources/venue.ex

```diff
@@ -135,6 +135,19 @@ defmodule CsGuide.Resources.Venue do
     |> Repo.insert()
   end
 
+  def update(%__MODULE__{} = item, attrs, postcode) do
+    item
+    |> __MODULE__.preload(__MODULE__.__schema__(:associations))
+    |> Map.put(:id, nil)
+    |> Map.put(:inserted_at, nil)
+    |> Map.put(:updated_at, nil)
+    |> __MODULE__.changeset(attrs)
+    |> Resources.put_many_to_many_assoc(attrs, :venue_types, CsGuide.Categories.VenueType, :name)
+    |> Resources.put_many_to_many_assoc(attrs, :drinks, CsGuide.Resources.Drink, :entry_id)
+    |> validate_postcode(postcode)
```

Comment:
Could not think of another way to implement this function but this one is pretty WET (not DRY) 🤔 

@Danwhy @nelsonic @SimonLab any suggestions would be appreciated 👍